crypto/x509/v3_purp.c

```diff
@@ -398,30 +417,34 @@ int x509v3_cache_extensions(X509 *x)
     }
     ERR_set_mark();
 
+    /* Cache the SHA1 digest of the cert */
     if (!X509_digest(x, EVP_sha1(), x->sha1_hash, NULL))
-            x->ex_flags |= EXFLAG_INVALID;
+        x->ex_flags |= EXFLAG_INVALID;
+    /*
+     * Note that the cert is marked invalid also on internal malloc failure
+     * or on failure of EVP_MD_fetch(), potentially called by X509_digest().
```

Comment:
Is this some sort of TODO? In any case, I think it makes more sense to put the comment above the code.

Reply:
It can be seen as a TODO because it would be better to improve the situation described.
I've moved the comment above the ` x->ex_flags |= EXFLAG_INVALID;`.